dashboard/app/serializers/lesson_serializer.rb

```diff
@@ -0,0 +1,28 @@
+# == Schema Information
+#
+# Table name: stages
+#
+#  id                :integer          not null, primary key
+#  name              :string(255)      not null
+#  absolute_position :integer
+#  script_id         :integer          not null
+#  created_at        :datetime
+#  updated_at        :datetime
+#  lockable          :boolean          default(FALSE), not null
+#  relative_position :integer          not null
+#  properties        :text(65535)
+#  lesson_group_id   :integer
+#  key               :string(255)
+#
+# Indexes
+#
+#  index_stages_on_script_id  (script_id)
+#
+
+# Serializer for the Lesson model. For now, only include fields which can
+# be edited on the lesson edit page, excluding any fields which are still
+# controlled by the script edit page (e.g. name and key).
+class LessonSerializer < ActiveModel::Serializer
+  attributes :overview, :student_overview, :assessment, :unplugged, :creative_commons_license,
+    :lockable, :purpose, :preparation, :announcements
+end
```

Comment:
this serializer is a bit bare / boring right now, but check out some examples of where we use a serializer to give different contents based on various factors in other parts of our code:
* rename / flatten deeply nested fields: https://github.com/code-dot-org/code-dot-org/blob/3627297ee386ff1520f9da15b301704b3188dacb/dashboard/app/serializers/api/v1/pd/workshop_serializer.rb#L48-L50
* different contents based on user's ability: https://github.com/code-dot-org/code-dot-org/blob/3627297ee386ff1520f9da15b301704b3188dacb/dashboard/app/serializers/api/v1/pd/workshop_serializer.rb#L52-L53 we might do something similar to make sure teacher-only markdown is only visible to verified teachers.

Reply:
it can also be used to serialize associations: https://github.com/rails-api/active_model_serializers/blob/v0.10.6/docs/general/serializers.md#associations